rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,564 @@
+            yield Chem.MolToMolBlock(self.moldict[molId], confId=confId)
+            
+    @property
+    def selectedMolNames(self):
+        """ Return the names of all selected molecules """
+        return self.rdkit_mol_select
+        
+    @property
+    def selectedConfIds(self):
+        """ Return the names of all selected molecules """
+        return self.rdkit_conf_select
+        
+    @property
+    def selectedMolecules(self):
+        """ Return the selected molecules """
+        return [self.moldict[name] for name in self.selectedMolNames]
+        
+    @property
+    def allConfIds(self):
+        """ Return the number of conformations - use the first selected molecule to determine """
+        nconfIds = self.selectedMolecules[0].GetNumConformers()
+        return list(range(nconfIds))
+        
+    @property
+    def getPropPrecalculated(self):
+        """ Return the precalculated properties """
+        if len(self.selectedMolNames) != 1:
+            return None
+        else:
+            mol = self.moldict[list(self.selectedMolNames)[0]]
+            if len(mol.GetPropNames()) == 0:
+                return 'Not found'
+            else:
+                return {prop: mol.GetProp(prop) for prop in mol.GetPropNames()}
+                
+    
+    
+class MolView3D(object):
+    """This function initiates required widgets and 3Dmol.js viewer
+    uid : an unique id for 3DMol.js viewer, each of the ipywidgets, and all the global objects
+    moldict : it is a molecule dictionary where is keys are string
+    protein : it is rdkit standard mol object
+    molStyle : representation type for moldict or small molecule
+    pStyle : protein representation style
+    ** the rest of the arguments are panels (user may or may not need those)
+    """
+    def __init__(self, 
+                 moldict = None, protein = None,
+                 molStyle = 'stick', pStyle='cartoon',
+                 molPanel = 'full',
+                 stylePanel = False, 
+                 labelPanel = False,
+                 propPanel = False, calculateProperties = False
+                 ):
+        
+        if molPanel not in ('full', 'minimal'):
+            raise KeyError('molPanel can be full or minimal')
+            
+        uid = str(time.time()).replace('.','')
+        
+        self.uid = uid
+        self.molStyle = molStyle
+        self.pStyle = pStyle
+        self.onStart = True
+        self.molPanel = molPanel # molPanel can be full or minimal
+        self.propPanel = propPanel
+        self.stylePanel = stylePanel
+        self.labelPanel = labelPanel
+        
+        wgData = globals().setdefault('wgData', {})
+        wgData[uid] = {}
+        
+        # Right hand panel (widgets)
+        
+        wgBox = []
+        
+        itemLayout=Layout(display='flex', flex_flow='row', justify_content='space-between')
+        
+        wgData[uid]['selectedMolsConfsView'] = HTML(description='', value='')
```

Comment:
This becomes
```
this.wgData['selectedMolsConfsView'] = HTML(description='', value='')
```
and so on.

The individual elements could now actually be individual fields
```
this.selectedMolsConfsView = HTML(description='', value='')
```
I'm not sure if I would do this now, because it might be more reasonable to extract the GUI controls code into a separate class. This class would then just link the 3D viewer to the GUI controls and manages their interaction. There are probably good reasons to do both ways.